extensions/binding/org.eclipse.smarthome.binding.fsinternetradio.test/src/test/java/org/eclipse/smarthome/binding/fsinternetradio/test/FSInternetRadioDiscoveryParticipantOSGiTest.java

```diff
@@ -0,0 +1,196 @@
+import org.eclipse.smarthome.binding.fsinternetradio.internal.FSInternetRadioDiscoveryParticipant;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.test.java.JavaOSGiTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.jupnp.model.ValidationException;
+import org.jupnp.model.meta.DeviceDetails;
+import org.jupnp.model.meta.ManufacturerDetails;
+import org.jupnp.model.meta.ModelDetails;
+import org.jupnp.model.meta.RemoteDevice;
+import org.jupnp.model.meta.RemoteDeviceIdentity;
+import org.jupnp.model.meta.RemoteService;
+import org.jupnp.model.types.DeviceType;
+import org.jupnp.model.types.UDN;
+
+/**
+ * OSGi tests for the {@link FSInternetRadioDiscoveryParticipant}.
+ *
+ * @author Mihaela Memova - Initial contribution
+ * @author Markus Rathgeb - Migrated from Groovy to pure Java test, made more robust
+ */
+public class FSInternetRadioDiscoveryParticipantOSGiTest extends JavaOSGiTest {
+    UpnpDiscoveryParticipant discoveryParticipant;
+
+    // default device variables used in the tests
+    DeviceType DEFAULT_TYPE = new DeviceType("namespace", "type");
+    String DEFAULT_UPC = "upc";
+    URI DEFAULT_URI = null;
+
+    // default radio variables used in most of the tests
+    private static final RemoteDeviceIdentity DEFAULT_RADIO_IDENTITY;
+    private static final URL DEFAULT_RADIO_BASE_URL;
+    String DEFAULT_RADIO_NAME = "HamaRadio";
+    static {
+        try {
+            DEFAULT_RADIO_IDENTITY = new RemoteDeviceIdentity(new UDN("radioUDN"), 60,
+                    new URL("http://radioDescriptiveURL"), null, null);
+            DEFAULT_RADIO_BASE_URL = new URL("http://radioBaseURL");
+        } catch (final MalformedURLException ex) {
+            throw new Error("Initialization error", ex);
+        }
+    }
+
+    /*
+     * The default radio is chosen from the {@link FrontierSiliconRadioDiscoveryParticipant}'s
+     * set of supported radios
+     */
+    String DEFAULT_RADIO_MANIFACTURER = "HAMA";
+    String DEFAULT_RADIO_MODEL_NAME = "IR";
+    String DEFAULT_RADIO_MODEL_DESCRIPTION = "IR Radio";
+    String DEFAULT_RADIO_MODEL_NUMBER = "IR100";
+    String DEFAULT_RADIO_SERIAL_NUMBER = "serialNumber123";
+
+    String RADIO_BINDING_ID = "fsinternetradio"; // taken from the binding.xml file
+    String RADIO_THING_TYPE_ID = "radio"; // taken from the thing-types.xml file
+    String DEFAULT_RADIO_THING_UID = String.format("%s:%s:%s", RADIO_BINDING_ID, RADIO_THING_TYPE_ID,
+            DEFAULT_RADIO_SERIAL_NUMBER);
+
+    @Before
+    public void setUp() {
+        discoveryParticipant = getService(UpnpDiscoveryParticipant.class, FSInternetRadioDiscoveryParticipant.class);
+        assertThat("The FSInternetRadioDiscoveryParticipant service cannot be found", discoveryParticipant,
```

Comment:
I agree, I don't think we need to add a custom string to the assertions at all.
If the assertion fails on a test, we need to look at the code to fix it.
And there is already "expected behavior" that is reported to not match.